src/esp_littlefs.c

```diff
@@ -1315,6 +1335,12 @@ static int vfs_littlefs_fstat(void* ctx, int fd, struct stat * st) {
         return -1;
     }
     file = efs->cache[fd];
+#if CONFIG_LITTLEFS_OPEN_DIR
```

Comment:
have you experimented if this block of code is necessary? I think `lfs_stat` works with directories, but I haven't experimented with it. If it works, it might provide more accurate `size` and maybe `type`.

Reply:
OK.

Reply:
Yes, this is needed in my customer's code, I will add more test case for this.

Reply:
is there any way to get a more accurate size than 0?

Reply:
Yes, I try to get the value by `lfs_stat`, it can get the real size(not 0).